src/util/read-pjson.ts

```diff
@@ -0,0 +1,64 @@
+import {cosmiconfig} from 'cosmiconfig'
+import {join} from 'node:path'
+
+import {PJSON} from '../interfaces'
+import {makeDebug} from '../logger'
+import {readJson} from './fs'
+
+const debug = makeDebug('read-pjson')
+
+/**
+ * Read the package.json file from a given path and merge with the config found by cosmiconfig.
```

Comment:
This is slightly misleading. It makes it sound like we are merging `pjson.oclif` config with with additional config discovered by `cosmicconfig`.

Reply:
Yeah I think "merge" as in, "put an `oclif` prop in the package.json"... I'll clarify it